### PR TITLE
docs: cleanup duplicated guides

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket:

### Changes included:

the common breaking change page was redundant with the client-specific, I've merged them and did a bit of cleanup for the GA

### CODE REVIEW

File: website/docs/clients/introduction.md [hunk 1]
```diff
@@ -13,8 +15,8 @@ Generated code in production can be find on repository of the clients.
 - [C#](https://github.com/algolia/algoliasearch-client-csharp/tree/next)
 - [Dart](https://github.com/algolia/algoliasearch-client-dart/)
 - [Go](https://github.com/algolia/algoliasearch-client-go/tree/next/)
-- [Java](https://github.com/algolia/algoliasearch-client-java/tree/next/)
 - [JavaScript](https://github.com/algolia/algoliasearch-client-javascript/tree/next/)
+- [Java](https://github.com/algolia/algoliasearch-client-java/tree/next/)
```
Comment (millotp) [inline website/docs/clients/introduction.md:19]: Alphabetically java is before javascript

**Review by millotp — COMMENTED** [review]
huge !

File: website/docs/clients/introduction.md (hunk 1 above)
Comment (shortcuts) [reply website/docs/clients/introduction.md:19]: because of the capital S it's before

Comment (millotp) [reply website/docs/clients/introduction.md:19]: yeah but nowhere else in the codebase we have that

**Review by millotp — APPROVED** [review]

**Pull request merged**